tools/hive-bigquery/bigquery_component.py

```diff
@@ -0,0 +1,725 @@
+                    "WHERE gcs_file_path='%s'" % (
+                        hive_table_model.tracking_table_name,
+                        bq_job_id, gcs_source_uri)
+            mysql_component.execute_transaction(query)
+            print_and_log(
+                "Updated BigQuery load job ID {} status TODO --> RUNNING for "
+                "file path {}".format(
+                    bq_job_id, gcs_source_uri))
+
+    def update_bq_job_status(self, mysql_component, gcs_component,
+                             hive_table_model, bq_table_model, gcs_bucket_name):
+        """Updates the status of running BigQuery load jobs
+
+        Queries the tracking table and fetches information about the load
+        jobs that are 'RUNNING 'and polls the job status. If the job has
+        finished successfully with no errors, updates the status as 'DONE'.
+        In case of job completion with errors, it updates the status to
+        'TODO' and increases the bq_job_retries count by 1. Waits for 1
+        minute and polls the job status again, until all the load jobs finish
+
+        Args:
+            mysql_component (:class:`MySQLComponent`): Instance of
+                MySQLComponent to connect to MySQL
+            gcs_component (:class:`GCSStorageComponent`): Instance of
+                GCSStorageComponent to do GCS operations
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+            gcs_bucket_name (str): GCS bucket name
+        """
+
+        # Uodate this value to increase the maximum number of load job retries
+        bq_load_job_max_retries = 3
+        print_and_log(
+            "Fetching information about BigQuery load jobs from tracking "
+            "table...")
+        query = "SELECT gcs_file_path,bq_job_id,bq_job_retries FROM %s WHERE " \
+                "bq_job_status='RUNNING'" % (
+                    hive_table_model.tracking_table_name)
+        results = mysql_component.execute_query(query)
+        if not results:
+            print_and_log(
+                "No BigQuery job is in RUNNING state. No values to update")
+
+        # Waits till all the load jobs finish
+        while results:
+            count = 0
+            for row in results:
+                gcs_file_path, bq_job_id, bq_job_retries = row
+                # Gets information about the running job
+                job = self.client.get_job(bq_job_id,
+                                          location=self.get_dataset_location(
+                                              bq_table_model.dataset_id))
+
+                if job.state == 'DONE':
+                    # Job finished successfully
+                    if job.errors is None:
+                        query = "UPDATE %s SET bq_job_status='DONE' WHERE " \
```

Comment:
Inconsistent use of String formatting. Let's switch this to new format if we can